### ISSUE TITLE
Tailwind 3 support

### ISSUE THREAD

**Issue by okkdev** [matheus23/elm-tailwind-modules#9]
Tailwind version 3 released yesterday: https://tailwindcss.com/blog/tailwindcss-v3

elm-tailwind-modules fails with:
```
(node:81442) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, stat 'generated in-memory'
    at Object.statSync (fs.js:1131:3)
    at trackModified (/Users/js/Documents/git/personal/elm-site-template/node_modules/tailwindcss/lib/lib/setupContextUtils.js:447:39)
    at Object.getContext (/Users/js/Documents/git/personal/elm-site-template/node_modules/tailwindcss/lib/lib/setupContextUtils.js:788:5)
    at /Users/js/Documents/git/personal/elm-site-template/node_modules/tailwindcss/lib/lib/setupTrackingContext.js:141:53
    at /Users/js/Documents/git/personal/elm-site-template/node_modules/tailwindcss/lib/processTailwindFeatures.js:37:11
    at plugins (/Users/js/Documents/git/personal/elm-site-template/node_modules/tailwindcss/lib/index.js:22:68)
    at LazyResult.runOnRoot (/Users/js/Documents/git/personal/elm-site-template/node_modules/postcss/lib/lazy-result.js:339:16)
    at LazyResult.runAsync (/Users/js/Documents/git/personal/elm-site-template/node_modules/postcss/lib/lazy-result.js:393:26)
    at LazyResult.async (/Users/js/Documents/git/personal/elm-site-template/node_modules/postcss/lib/lazy-result.js:221:30)
    at LazyResult.then (/Users/js/Documents/git/personal/elm-site-template/node_modules/postcss/lib/lazy-result.js:206:17)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:81442) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:81442) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by matheus23** [comment]
If someone is up for the challenge, I'd love to merge a tailwind 3 support PR.
I haven't done any research yet and I think that's a first step. We need to figure out whether we can even support tailwind's 3's "JIT" mechanism - or turn it off somehow.

**Comment by Erudition** [comment]
Super excited for this!

**Comment by okkdev** [comment]
This might be interesting: https://tailwindcss.com/blog/standalone-cli

**Comment by ocharles** [comment]
That looks good, but I still think the JIT part is going to be hard - I think Tailwind now needs to know up front exactly which classes you want to use

**Comment by okkdev** [comment]
Yeah, there's also no way to disable JIT. There's only the pattern safelist to force generation https://github.com/tailwindlabs/tailwindcss/pull/5511 but that doesn't solve the JIT incompatibility.

**Comment by matheus23** [comment]
Thinking out loud:
There has to be a way to list all css rules that tailwind 3 generates given a certain config, because I'm sure the tailwind folks want to have autocomplete, right?
Looking at the tailwind-intellisense project, it seems that there's an issue about it not working with V3, so I may be wrong. But eventually, they'll want that is what I'm thinking, so there will have to be *some* way eventually.

If that exited, we could use that to generate all class names and then possibly query the tailwind engine to give us the necessary styles.

**Comment by Clindbergh** [comment]
> Looking at the tailwind-intellisense project, it seems that there's an issue about it not working with V3

Are you talking about [this issue](https://github.com/tailwindlabs/tailwindcss-intellisense/issues/448)? It's resolved now.

**Comment by matheus23** [comment]
Thanks for your work on this @leojpod :pray:

**Comment by matheus23** [comment]
I just published version 0.4.0 of this package which should have tailwind v3 support.

**Comment by leojpod** [comment]
You're welcome :)
